pandas/core/dtypes/base.py

```diff
@@ -213,7 +213,7 @@ def construct_array_type(cls) -> type_t[ExtensionArray]:
         raise NotImplementedError
 
     @classmethod
-    def construct_from_string(cls, string: str):
+    def construct_from_string(cls, string: str) -> ExtensionDtype:
```

Comment:
can/should we do the `cls: Type[ExtensionDtypeT]` thing here?

Reply:
Done there (and in other class methods) in latest commit.